### PR TITLE
qbd-direct post onboarding configuration export settings html

### PR DESCRIPTION
**Pull request by DhaaraniCIT**

### CODE REVIEW

**Comment by DhaaraniCIT** [comment]
Screenshot 2024-09-10 at 3 48 38 PM

Screenshot 2024-09-10 at 3 48 42 PM

File: src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-export-settings/qbd-direct-export-settings.component.html [hunk 1]
```diff
@@ -1 +1,178 @@
...
+            </app-configuration-step-header>
+        </div>
+        <form [formGroup]="exportSettingsForm">
+            <div class="tw-p-24-px"
+                [ngClass]="{'tw-pt-0': !brandingFeatureConfig.featureFlags.exportSettings.reimbursableExpenses}">
+                <div *ngIf="brandingFeatureConfig.featureFlags.exportSettings.reimbursableExpenses" class="tw-mb-16-px">
+                    <div class="tw-rounded-lg tw-border-border-tertiary tw-border tw-bg-bg-tertiary-lighter">
+                        <app-configuration-toggle-field [form]="exportSettingsForm"
+                            [label]="'Export Reimbursable Expenses'"
+                            [subLabel]="'Enable this to export the reimbursable expenses from ' + brandingConfig.brandName + '. If not enabled, any <b>out-of-pocket</b> expenses will not be exported to Quickbooks Desktop.'"
```
Comment (ashwin1111) [inline src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-export-settings/qbd-direct-export-settings.component.html:20]: all contents to come from ts file

File: src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-export-settings/qbd-direct-export-settings.component.html [hunk 2]
```diff
@@ -1 +1,178 @@
...
+                [showSyncButton]="isOnboarding" [appName]="appName" (refreshDimension)="refreshDimensions()">
+            </app-configuration-step-header>
+        </div>
+        <form [formGroup]="exportSettingsForm">
+            <div class="tw-p-24-px"
+                [ngClass]="{'tw-pt-0': !brandingFeatureConfig.featureFlags.exportSettings.reimbursableExpenses}">
+                <div *ngIf="brandingFeatureConfig.featureFlags.exportSettings.reimbursableExpenses" class="tw-mb-16-px">
+                    <div class="tw-rounded-lg tw-border-border-tertiary tw-border tw-bg-bg-tertiary-lighter">
+                        <app-configuration-toggle-field [form]="exportSettingsForm"
+                            [label]="'Export Reimbursable Expenses'"
```
Comment (ashwin1111) [inline src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-export-settings/qbd-direct-export-settings.component.html:19]: all contents to come from ts file

File: src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-export-settings/qbd-direct-export-settings.component.html [hunk 3]
```diff
@@ -1 +1,178 @@
...
+                            [isSectionHeader]="true">
+                        </app-configuration-toggle-field>
+                    </div>
+
+                    <div *ngIf="exportSettingsForm.value?.reimbursableExpense">
+                        <div
+                            class="tw-mt-16-px tw-bg-white tw-border tw-border-solid tw-border-border-tertiary tw-rounded-12-px">
+                            <app-configuration-select-field [form]="exportSettingsForm" [isFieldMandatory]="true"
+                                [mandatoryErrorListName]="'export module'"
+                                [label]="'How should the expenses be exported?'"
```
Comment (ashwin1111) [inline src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-export-settings/qbd-direct-export-settings.component.html:31]: all contents to come from ts file

File: src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-export-settings/qbd-direct-export-settings.component.html [hunk 4]
```diff
@@ -1 +1,178 @@
...
+                                [subLabel]="'Choose the type of transaction in Quickbooks Desktop to export your ' + brandingConfig.brandName +' expenses'"
```
Comment (ashwin1111) [inline src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-export-settings/qbd-direct-export-settings.component.html:32]: all contents to come from ts file

Comment (ashwin1111) [reply src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-export-settings/qbd-direct-export-settings.component.html:32]: pls take care of this everywhere

File: src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-export-settings/qbd-direct-export-settings.component.html [hunk 5]
```diff
@@ -1 +1,178 @@
...
+                            [mandatoryErrorListName]="'export date'"
+                            [label]="'Set the ' + (exportSettingsForm.value.reimbursableExportType | snakeCaseToSpaceCase | titlecase) + ' date as'"
+                            [subLabel]="'The selected type of date will reflect in the expenses exported from ' + brandingConfig.brandName + ' to Quickbooks Desktop.'"
+                            [options]="reimbursableExpenseGroupingDateOptions" [iconPath]="'calendar'"
+                            [placeholder]="'Select the date of the ' + (exportSettingsForm.value.reimbursableExportType | snakeCaseToSpaceCase | titlecase)"
+                            [formControllerName]="'reimbursableExportDate'">
+                        </app-configuration-select-field>
+                    </div>
+                </div>
+                <div *ngIf="brandingFeatureConfig.featureFlags.exportSettings.reimbursableExpenses" class="tw-rounded-lg tw-border-border-tertiary tw-border tw-bg-bg-tertiary-lighter">
```
Comment (ashwin1111) [inline src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-export-settings/qbd-direct-export-settings.component.html:81]: why is ng if required?

File: src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-export-settings/qbd-direct-export-settings.component.html (hunk 1 above)
Comment (DhaaraniCIT) [reply src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-export-settings/qbd-direct-export-settings.component.html:20]: for reimbursement section also we need to have test come from ts, for ccc we have

Comment (DhaaraniCIT) [reply src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-export-settings/qbd-direct-export-settings.component.html:20]: for reimbursement section also the text should come from ts file, like ccc?

Comment (ashwin1111) [reply src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-export-settings/qbd-direct-export-settings.component.html:20]: yes, everywhere - content across all components for qbd

Comment (DhaaraniCIT) [reply src/app/integrations/qbd-direct/qbd-direct-shared/qbd-direct-export-settings/qbd-direct-export-settings.component.html:20]: cool, ok

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**